src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/compute/v4_9/CloudStackComputePluginTest.java

```diff
@@ -1,894 +1,1426 @@
 
-        PowerMockito.mockStatic(CloudStackUrlUtil.class);
-        PowerMockito.when(CloudStackUrlUtil.createURIBuilder(Mockito.anyString(), Mockito.anyString())).thenCallRealMethod();
+        GetVirtualMachineResponse response = Mockito.mock(GetVirtualMachineResponse.class);
+        Mockito.doThrow(new FogbowException()).when(this.plugin).requestGetVirtualMachine(
+                Mockito.eq(request), Mockito.eq(cloudStackUser));
 
-        String fakeImageId = "fake-image-id";
+        // exercise
+        try {
+            this.plugin.doGetInstance(request, cloudStackUser);
+        } catch (Exception e) {
+            Mockito.verify(this.plugin, Mockito.times(TestUtils.NEVER_RUN)).buildComputeInstance(
+                    Mockito.eq(response), Mockito.eq(cloudStackUser));
+        }
+    }
 
-        UserData fakeUserData = new UserData("fakeuserdata", CloudInitUserDataBuilder.FileType.CLOUD_CONFIG, "fake-tag");
-        String fakeUserDataString = Base64.getEncoder().encodeToString(
-                fakeUserData.getExtraUserDataFileContent().getBytes("UTF-8"));
-        ArrayList<UserData> userData = new ArrayList<>();
-        userData.add(fakeUserData);
+    // test case: successfully case
+    @Test
+    public void testDoRequestInstance() throws FogbowException {
+        // set up
+        CloudStackUser cloudStackUser = CLOUD_STACK_USER;;
+        ComputeOrder computeOrder = Mockito.mock(ComputeOrder.class);
+        GetAllServiceOfferingsResponse.ServiceOffering serviceOffice =
+                Mockito.mock(GetAllServiceOfferingsResponse.ServiceOffering.class);
+        DeployVirtualMachineRequest request = Mockito.mock(DeployVirtualMachineRequest.class);
+        GetAllDiskOfferingsResponse.DiskOffering diskOffering =
+                Mockito.mock(GetAllDiskOfferingsResponse.DiskOffering.class);
+
+        String instanceIdExpected = "instanceId";
+        DeployVirtualMachineResponse response = Mockito.mock(DeployVirtualMachineResponse.class);
+        Mockito.when(response.getId()).thenReturn(instanceIdExpected);
+        Mockito.doReturn(response).when(this.plugin)
+                .requestDeployVirtualMachine(Mockito.eq(request), Mockito.eq(cloudStackUser));
+
+        Mockito.doNothing().when(this.plugin).updateComputeOrder(Mockito.eq(computeOrder),
+                Mockito.eq(serviceOffice), Mockito.eq(diskOffering));
+
+        String instanceId = this.plugin.doRequestInstance(
+                request, serviceOffice, diskOffering, computeOrder, cloudStackUser);
+
+        Assert.assertEquals(instanceIdExpected, instanceId);
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).updateComputeOrder(
+                Mockito.eq(computeOrder), Mockito.eq(serviceOffice), Mockito.eq(diskOffering));
+    }
 
-        String fakeNetworkIdsString = this.defaultNetworkId + "," + FAKE_NETWORK_ID;
+    // test case: It's throw an exception in the request to the cloud and the method never update the order
```

Comment:
I think you mean: When an exception is thrown while cloud requesting, the method won't update the order.. or some thing like that, try to improve the test case's description, please.

Reply:
Done